File: problems/baekjoon/p2588.cpp
/**
 * (211001) 곱셈
 * https://www.acmicpc.net/problem/2588
 * 
 * 풀이
 * (1) 세 자리 수 정수 두개를 각각 변수 A, B에 대입한다.
 * (2) A * B의 값을 구해서 저장해 둔다.
 * (3) B를 뒤에서 부터 한 자리씩 가져와서 A와 곱하고 결과를 출력한다.
 * (4) 2번에서 구해놓은 값을 출력한다.
 */

#include <cstdio>

int main() {
  int A, B;
  scanf("%d", &A);
  scanf("%d", &B);

  int result = A * B;

  int current;
  while (B > 0) {
    current = B % 10;
    printf("%d\n", A * current);

    B /= 10;
  }

  printf("%d\n", result);

  return 0;
}

File: problems/csharp_baekjoon/p2588.cs
/*
 * (220413) 곱셈
 * https://www.acmicpc.net/problem/2588
 *
 * [풀이]
 * 정수A 를 뒤에서 부터 한자리씩 읽기 위해서는
 *
 * Console.WriteLine(A % 10);
 * A /= 10
 *
 * 위 두 코드를 A가 0 이하가 될 때까지 반복하면 된다.
 */

using System;
using System.Collections.Generic;

namespace Baekjoon
{
  class Program
  {
    static void Main()
    {
      int A = int.Parse(Console.ReadLine());
      int B = int.Parse(Console.ReadLine());

      List<int> nums = new List<int>();
      int counter = 0, current;

      while (B > 0) {
        current = A * (B % 10);
        Console.WriteLine(current);
        
        nums.Add(current * (int)Math.Pow(10, counter++));
        
        B /= 10;
      }

      int answer = 0;

      foreach (int num in nums) {
        answer += num;
      }

      Console.WriteLine(answer);
    }
  }
}
